### PR TITLE
Fixed EKF height innovation gate on the ground

### LINKED ISSUE #11903: Temporary baro error causes large attitude error
A short term (less than 1 second) error on a barometer can cause the EKF attitude to go wildly off. This is reproducible in bench tests, and in EKF replay.

A log showing the issue is here:
http://uav.tridgell.net/Paul/baro-error-EKF-error.bin
the log was recorded with LOG_REPLAY=1 and LOG_DISARMED=1. Running it through replay does reproduce the error.

image
image

The issue is also reproducible in SITL by using SIM_BARO_GLITCH. Use a copter disarmed on the ground, and then set SIM_BARO_GLITCH to -14000 for 1 second, setting it back to 0 afterwards. The attitude goes crazy:
image

### PR DESCRIPTION
**Pull request by tridge**
The height innovation gate when on the ground without GPS was unlimited, as the innovation limit check was skipped with AID_NONE and onGround.
This meant that a single extreme baro reading (due for example to a brief SPI bus error) could completely destabilise the filter as it fused in a value beyond what the EKF can cope with.
This change increases the test ratio limit to 3.0 from 1.0 when onGround with AIDING_NONE, which still gives us increased chance to fuse the baro without allowing the filter to destabilise.
Note that the value of 3.0 is pretty arbitrary. I'm hoping Paul can suggest a suitable value.
Fixes #11903

### CODE REVIEW

File: libraries/AP_NavEKF2/AP_NavEKF2_PosVelFusion.cpp [hunk 1]
```diff
@@ -602,10 +602,15 @@ void NavEKF2_core::FuseVelPosNED()
             varInnovVelPos[5] = P[8][8] + R_OBS_DATA_CHECKS[5];
             // calculate the innovation consistency test ratio
             hgtTestRatio = sq(innovVelPos[5]) / (sq(MAX(0.01f * (float)frontend->_hgtInnovGate, 1.0f)) * varInnovVelPos[5]);
-            // fail if the ratio is > 1, but don't fail if bad IMU data
-            hgtHealth = ((hgtTestRatio < 1.0f) || badIMUdata);
+
+            // when on the ground we accept a larger test ratio
```
Comment (priseborough) [inline libraries/AP_NavEKF2/AP_NavEKF2_PosVelFusion.cpp]: Would be worth adding a few words to explain why we are doing this, eg

// when on ground we accept a larger test ratio to allow the filter to handle 
// large switch on IMU bias errors without rejecting the height sensor

File: libraries/AP_NavEKF3/AP_NavEKF3_PosVelFusion.cpp [hunk 2]
```diff
@@ -564,10 +564,15 @@ void NavEKF3_core::FuseVelPosNED()
             varInnovVelPos[5] = P[9][9] + R_OBS_DATA_CHECKS[5];
             // calculate the innovation consistency test ratio
             hgtTestRatio = sq(innovVelPos[5]) / (sq(MAX(0.01f * (float)frontend->_hgtInnovGate, 1.0f)) * varInnovVelPos[5]);
+
+            // when on the ground we accept a larger test ratio
```
Comment (priseborough) [inline libraries/AP_NavEKF3/AP_NavEKF3_PosVelFusion.cpp]: Same comment as AP_NavEKF2

File: libraries/AP_NavEKF2/AP_NavEKF2_PosVelFusion.cpp (hunk 1 above)
Comment (tridge) [reply libraries/AP_NavEKF2/AP_NavEKF2_PosVelFusion.cpp]: will do. I didn't actually know why it was letting them through unchecked. Now I know I can add a comment :-)

File: libraries/AP_NavEKF3/AP_NavEKF3_PosVelFusion.cpp (hunk 2 above)
Comment (tridge) [reply libraries/AP_NavEKF3/AP_NavEKF3_PosVelFusion.cpp]: fixed

**Pull request merged**